eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -114,6 +115,8 @@ object PaymentReceived {
 
 }
 
+case class PaymentMetadataReceived(paymentHash: ByteVector32, paymentMetadata: ByteVector)
```

Comment:
You can also add it to the `PaymentReceived` defined above instead of creating a new event just for the metadata.

Reply:
I tried, but it is much more invasive...and I'm not sure in the long run we will want it in `PaymentReceived` (which happens after the payment has been accepted and fulfilled), but instead before we accept it.